apps/gdalwarp_lib.cpp

```diff
@@ -3873,22 +3873,72 @@ static GDALDatasetH GDALWarpCreateOutput(
                 }
             }
 
-            psInfo->pfnTransform(hTransformArg, TRUE, nPoints, &padfX[0],
-                                 &padfY[0], &padfZ[0], &pabSuccess[0]);
+            bool transformedToSrcCRS{false};
+
+            GDALGenImgProjTransformInfo *psTransformInfo{
+                static_cast<GDALGenImgProjTransformInfo *>(hTransformArg)};
+
+            // If a transformer is available, use an extent that covers the
+            // target extent instead of the real source image extent
+            if (psTransformInfo && psTransformInfo->pReprojectArg)
+            {
+                const GDALReprojectionTransformInfo *psRTI =
+                    static_cast<const GDALReprojectionTransformInfo *>(
+                        psTransformInfo->pReprojectArg);
+                if (psRTI && psRTI->poReverseTransform)
+                {
+
+                    // Compute new geotransform from transformed target extent
+                    double adfGeoTransform[6];
+                    if (GDALGetGeoTransform(hSrcDS, adfGeoTransform) == CE_None)
```

Comment:
below code assumes that gt[2] and gt[5] are 0. We should check this assumption here

Reply:
did you mean gt[2] and gt[4] ?

Reply:
done